skopt/wrappers/search_cv.py

```diff
@@ -0,0 +1,490 @@
+            # validation splits for some particular parameter values
+            I = slice(i*splits, (i+1)*splits, 1)
+
+            # record results for particular parameters point
+            self.cv_results_['params'].append(
+                self._skopt_to_dict(param_space, params)
+            )
+            for name, result in out_and_name:
+                for split, data in enumerate(result[I]):
+                    self.cv_results_['split'+str(split)+"_"+name].append(data)
+                self.cv_results_['mean_' + name].append(np.mean(result[I]))
+                self.cv_results_['std_' + name].append(np.std(result[I]))
+
+            # update index of best parameters if necessary
+            score = np.mean(test_scores[I])
+            if self.best_index_ is None or score > self.best_score_:
+                self.best_index_ = len(self.cv_results_['params'])-1
+
+            # feed the point and objective back into optimizer
+            optimizer.tell(params, -score)
+
+        # fit the best model if necessary
+        if self.refit:
+            self._fit_best_model(X, y)
+
+    def fit(self, X, y=None, groups=None):
+        """Run fit on the estimator with randomly drawn parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Training vector, where n_samples in the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples] or [n_samples, n_output]
+            Target relative to X for classification or regression;
+
+        groups : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+        """
+
+        # check if the list of parameter spaces is provided. If not, then
+        # only step in manual mode can be used.
+
+        if self.search_spaces is None:
+            raise ValueError(
+                "None search space is only supported with manual usage of "
+                "`step` method. Please provide list of search spaces "
+                " at initialization stage in order to be able to use"
+                "`fit` method."
+            )
+
+        n_jobs = self.n_jobs
+
+        # account for case n_jobs < 0
+        if n_jobs < 0:
+            n_jobs = max(1, cpu_count() + n_jobs + 1)
+
+        for elem in self.search_spaces:
+
+            # if tuple: (dict: search space, int: n_iter)
+            if isinstance(elem, tuple):
+                psp, n_iter = elem
+            # if dict: represents search space
+            elif isinstance(elem, dict):
+                psp, n_iter = elem, self.n_iter
+            else:
+                raise ValueError("Unsupported type of parameter space. "
+                                 "Expected tuple or dict, got " + str(elem))
+
+            # do the optimization for particular search space
+            while n_iter:
```

Comment:
`bool(-3) == True` so this won't stop if `n_iter` isn't dividable by `n_jobs`. I think we should check if there are enough iterations left to sample `n_jobs` more points, if not reduce it to the number left and then stop the loop. (and add a unit test to check this works)